src/main/java/jdbchomework/dao/jdbc/ProjectsJdbcDao.java

```diff
@@ -2,11 +2,58 @@
 
 import jdbchomework.dao.model.ProjectsDao;
 import jdbchomework.entity.Project;
-import org.hibernate.SessionFactory;
+import org.slf4j.LoggerFactory;
 
-public class ProjectsJdbcDao extends AbstractDao<Project> implements ProjectsDao {
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
 
-    public ProjectsJdbcDao(SessionFactory sessionFactory, String entity, Class<Project> clazz) {
-        super(sessionFactory, entity, clazz);
+public class ProjectsJdbcDao extends AbstractJdbcDao<Project> implements ProjectsDao {
+    private static final String ERROR_MESSAGE = "Cannot connect to DB";
+
+    private static org.slf4j.Logger log = LoggerFactory.getLogger(ProjectsJdbcDao.class);
+
+    public ProjectsJdbcDao(Connection connection, String table, String column) {
+        super(connection, table, column);
+    }
+
+    @Override
+    public void add(Project toAdd) {
+        String name = toAdd.getName();
+        int cost = toAdd.getCost();
+        try (PreparedStatement statement = connection
+                .prepareStatement("INSERT INTO projects (name, cost) VALUES (?,?)")) {
+            statement.setString(1, name);
+            statement.setInt(2, cost);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            log.error(ERROR_MESSAGE, e);
+            throw new RuntimeException(ERROR_MESSAGE, e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)